Petshop-iOS/Petshop-iOS/Views/CardView.swift

```diff
@@ -0,0 +1,171 @@
+    
+    private var originalCenter: CGPoint = .zero
+    private var isDraggingCard: Bool = false
+    
+    public var title: String = "" {
+        didSet {
+            cardTitle.text = title
+        }
+    }
+    
+    public var imageName: String = "" {
+        didSet {
+            cardImage.image = UIImage(named: imageName)
+        }
+    }
+    
+    public var descriptionText: String = "" {
+        didSet {
+            cardDescription.text = descriptionText
+        }
+    }
+    
+    public var button: UIButton {
+        return forwardButton
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupUI()
+        setupPanGesture()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("(init(coder:) has not been implemented")
+    }
+    
+    func configure(withTitle title: String, description: NSAttributedString, image: UIImage?, button: UIButton?) {
+            cardImage.image = image
+            cardTitle.text = title
+            cardDescription.attributedText = description
+        
+        if let button = button {
+                forwardButton.setTitle(button.title(for: .normal), for: .normal)
+                forwardButton.backgroundColor = button.backgroundColor
+                forwardButton.setTitleColor(button.titleColor(for: .normal), for: .normal)
+                forwardButton.layer.cornerRadius = button.layer.cornerRadius
+        }
+    }
+        
+        private func setupUI() {
+            backgroundColor = .primaryBackgroundColor
+            layer.cornerRadius = 12
+            
+            cardImageContainer.backgroundColor = .primaryBackgroundColor
+            cardImageContainer.layer.cornerRadius = 40
+            cardImageContainer.layer.masksToBounds = true
+            cardImageContainer.translatesAutoresizingMaskIntoConstraints = false
+            addSubview(cardImageContainer)
+            
+            cardImage.contentMode = .scaleAspectFill
+            cardImage.clipsToBounds = true
+            cardImage.layer.cornerRadius = 14
+            cardImage.translatesAutoresizingMaskIntoConstraints = false
+            cardImageContainer.addSubview(cardImage)
+            
+            cardTitle.textAlignment = .center
+            cardTitle.font = UIFont.boldSystemFont(ofSize: 24)
+            cardTitle.textColor = .titleColor
+            cardTitle.translatesAutoresizingMaskIntoConstraints = false
+            addSubview(cardTitle)
+            
+            cardDescription.textAlignment = .center
+            cardDescription.font = UIFont.systemFont(ofSize: 16)
+            cardDescription.textColor = .textColor
+            cardDescription.numberOfLines = 0
+            cardDescription.translatesAutoresizingMaskIntoConstraints = false
+            addSubview(cardDescription)
+            
+            forwardButton.setTitle("", for: .normal)
+            forwardButton.setTitleColor(.primaryBackgroundColor, for: .normal)
```

Comment:
Aqui voce está setando a cor do texto (Title) do botão ou do background? Revise meu comentário sobre duplicação de variáveis com mesma cor se for o caso.